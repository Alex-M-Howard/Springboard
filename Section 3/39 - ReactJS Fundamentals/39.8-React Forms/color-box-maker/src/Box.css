.Box{
    display: flex;
    margin: 5px;
}

.Box-Remove{
    height: 25px;
    width: 25px;
    border-radius: 100%;
    background-color: red;
    color: white;
}