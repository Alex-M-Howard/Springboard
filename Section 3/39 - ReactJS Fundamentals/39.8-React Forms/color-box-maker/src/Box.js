import React from 'react';
import './Box.css'

const Box = ({id, color='blue', width='200', height='200', handleRemove}) => {
  const remove = () => handleRemove(id);

    return (
      <div className="Box">
        <div
          className="Box-Color"
          data-testid="box-element"
          style={{
            backgroundColor: color,
            width: `${width}px`,
            height: `${height}px`,
          }}
          
        ></div>
        <button className="Box-Remove " onClick={remove}>X</button>
      </div>
    );
}

export default Box;