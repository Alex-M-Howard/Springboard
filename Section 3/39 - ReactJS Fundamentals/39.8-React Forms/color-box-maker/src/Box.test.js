import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box.js";

it("should render", () => {
  render(<Box />);
});

it("should render manual box", () => {
  render(<Box id={5} color='blue' width='200' height='200' handleRemove />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
