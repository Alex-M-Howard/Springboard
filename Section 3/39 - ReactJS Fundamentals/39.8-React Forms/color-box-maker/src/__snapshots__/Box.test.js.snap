// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-Color"
      data-testid="box-element"
      style="background-color: blue; width: 200px; height: 200px;"
    />
    <button
      class="Box-Remove "
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
